### PR TITLE
Add analytics extension with GoatCounter support

### PR DESCRIPTION
**Pull request by maximlt**
Implements https://github.com/holoviz-dev/nbsite/issues/294

Inspired from the conda extension and adapted having in mind we should upstream it one day to the pydata sphinx theme.

In their *conf.py* HoloViz sites will have to register the extension `nbsite.analytics` and set:

```python
nbsite_analytics = {
    'goatcounter_holoviz': True,
}
```

---

One issue I've seen in the generated HTML is that the `app.add_js_file(None, body=...)` call seems to be inserted twice. I don't think it's an issue with the code written in this PR, `app.add_js_file(None, body=...)` is called just once as far as I can see. Internal Sphinx issue? I don't know whether this is going to cause any problem or not in practice. This PR should build nbsite's dev site so we'll see immediately if we get data on GoatCounter's dashboard.

```html
<script>
    window.goatcounter = {
        path: function(p) { return location.host + p }
    }
</script>
<script async="async" data-goatcounter="https://holoviz.goatcounter.com" src="_static/js/goatcounter.js?v=f53f84b8"></script>
<script>
    window.goatcounter = {
        path: function(p) { return location.host + p }
    }
</script>
```

### CODE REVIEW

**Comment by maximlt** [comment]
Actually the dev site built from this PR doesn't suffer from this double JS injection I described above. Not sure what's going on on my local system.

Source of the deployed site:
image

File: nbsite/analytics/__init__.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+                body = (
+                    "\n    window.goatcounter = {\n"
+                    "        path: function(p) { return " + domain + " + p }\n"
+                    "    }\n"
+                )
+                app.add_js_file(None, body=body)
+        app.add_js_file(
+            "js/goatcounter.js",
+            **{"loading_method": "async", "data-goatcounter": goatcounter_url}
+        )
```
Comment (droumis) [inline nbsite/analytics/__init__.py:38]:
```suggestion
    # goatcounter_holoviz is specific to HoloViz, adjust before upstreaming
    if nbsite_analytics.get('goatcounter_holoviz', False):
        goatcounter_url = 'https://holoviz.goatcounter.com/count'
        goatcounter_domain = nbsite_analytics.get('goatcounter_domain', 'auto')
        # See https://www.goatcounter.com/help/domains
        domain = "location.host" if goatcounter_domain == "auto" else f"'{goatcounter_domain}'"

        body = (
            "\n    window.goatcounter = {\n"
            "        path: function(p) { return " + domain + " + p }\n"
            "    }\n"
        )
        app.add_js_file(None, body=body)
        app.add_js_file(
            "js/goatcounter.js",
            **{"loading_method": "async", "data-goatcounter": goatcounter_url}
        )
```

If I'm following the logic, we are hardcoding the URL, so this should be equivalent without the extra conditionals.

**Review by droumis — APPROVED** [review]
LGTM. Simplification of the logic suggested but not required.

File: nbsite/analytics/__init__.py (hunk 1 above)
Comment (maximlt) [reply nbsite/analytics/__init__.py:38]: You're right that for us your changes would work fine but I prefer to keep some flexibility in the case we in the end decide (or are forced for some reasons) to have multiple GoatCounter accounts.

**Comment by maximlt** [comment]
Thanks for the review! I'm going to merge and tag.

**Pull request merged**